### PR TITLE
cherry pick #4734 and # 4799

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
Why add DNM label?

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -825,7 +824,16 @@ impl Peer {
                         "{} becomes leader and lease expired time is {:?}",
                         self.tag, self.leader_lease
                     );
-                    self.heartbeat_pd(worker)
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: Why skip heartbeating PD?

File: src/raftstore/store/peer.rs [hunk 2]
```diff
@@ -608,14 +608,14 @@ impl Peer {
     }
 
     #[inline]
-    fn send<T, I>(&mut self, trans: &T, msgs: I, metrics: &mut RaftMessageMetrics) -> Result<()>
+    fn send<T, I>(&mut self, trans: &mut T, msgs: I, metrics: &mut RaftMessageMetrics)
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: Why mut?

Comment (hicqu) [reply src/raftstore/store/peer.rs]: Otherwise can't pass compiling.

Comment (BusyJay) [reply src/raftstore/store/peer.rs]: Why?

Comment (hicqu) [reply src/raftstore/store/peer.rs]: `send_raft_msg` in master needs a mut `T`. However I found we can remove the `mut`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 All notable changes to this project are documented in this file.
 See also [TiDB Changelog](https://github.com/pingcap/tidb/blob/master/CHANGELOG.md) and [PD Changelog](https://github.com/pingcap/pd/blob/master/CHANGELOG.md).
 
+## [Unreleased]
+- Ignore errors when sending raft message [4734](https://github.com/tikv/tikv/pull/4734)
+- More check for leader transferring during configuration changes [4799](https://github.com/tikv/tikv/pull/4853)
```
Comment (BusyJay) [inline CHANGELOG.md]: Wrong url.

**Comment by hicqu** [comment]
/run-all-tests tidb=release-2.1 pd=release-2.1 tidb-test=release-2.1 tidb-private-test=release-2.1

**Comment by hicqu** [comment]
/run-all-tests tidb=release-2.1 pd=release-2.1 tidb-test=release-2.1 tidb-private-test=release-2.1

**Comment by hicqu** [comment]
There are some problems about CI, I guess it's because #4734 has too many differences from release-2.1, and when I try to resolve those conflicts, some mistakes are introduced.

**Pull request closed without merging**